src/plugins/intel_gpu/tests/unit/shape_infer/matmul_si_test.cpp

```diff
@@ -120,6 +120,24 @@ INSTANTIATE_TEST_SUITE_P(smoke, fully_connected_test,
             layout{ov::PartialShape::dynamic(3), data_types::f32, format::bfyx},
             data_types::f32, false, false,
             layout{ov::PartialShape::dynamic(3), data_types::f32, format::bfyx}
+        },
+        {
+            layout{ov::PartialShape{10, 10, 10, 1024}, data_types::i8, format::bfyx},
+            layout{ov::PartialShape{1000, 1024}, data_types::i8, format::bfyx},
+            data_types::f16, false, false,
+            layout{ov::PartialShape{10, 10, 10, 1000}, data_types::f16, format::bfyx}
+        },
+        {
+            layout{ov::PartialShape{10, 10, 10, 1024}, data_types::f32, format::bfyx},
+            layout{ov::PartialShape{1000, 1024}, data_types::f32, format::bfyx},
+            data_types::f32, false, false,
+            layout{ov::PartialShape{10, 10, 10, 1000}, data_types::f32, format::bfyx}
+        },
+        {
+            layout{ov::PartialShape::dynamic(4), data_types::f32, format::bfyx},
+            layout{ov::PartialShape::dynamic(2), data_types::f32, format::bfyx},
+            data_types::f32, false, false,
+            layout{ov::PartialShape::dynamic(4), data_types::f32, format::bfyx}
```

Comment:
These test params do not cover your changes in `calc_output_layout`. Please add function test case with 4d input + 2d weight shapes for checking shape inference and accuracy together.